test/e2e/preset_test.go

```diff
@@ -608,6 +612,182 @@ func validateWorkspaceReadiness(workspaceObj *kaitov1alpha1.Workspace) {
 	})
 }
 
+// Create a temporary debug pod with curl if it doesn't already exist
+func createCurlDebugPod(namespace string) error {
+	By("Creating a temporary curl debug pod")
+
+	existingPod := &v1.Pod{}
+	err := utils.TestingCluster.KubeClient.Get(context.TODO(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      curlPodName,
+	}, existingPod)
+
+	if err == nil {
+		By(fmt.Sprintf("Debug pod %s already exists", curlPodName))
+		return nil
+	} else if !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to check existing pod: %v", err)
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      curlPodName,
+			Namespace: namespace,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "curl-container",
+					Image: "curlimages/curl",
+					Command: []string{
+						"sleep", "3600", // Keeps the pod running for long enough
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+
+	// Create the pod
+	err = utils.TestingCluster.KubeClient.Create(context.TODO(), pod)
+	if err != nil {
+		return fmt.Errorf("failed to create curl debug pod: %v", err)
+	}
+
+	// Wait for the pod to be Running
+	Eventually(func() bool {
+		fetchedPod := &v1.Pod{}
+		err := utils.TestingCluster.KubeClient.Get(context.TODO(), client.ObjectKey{
+			Namespace: namespace,
+			Name:      curlPodName,
+		}, fetchedPod)
+		if err != nil {
+			return false
+		}
+		return fetchedPod.Status.Phase == v1.PodRunning
+	}, 60*time.Second, 2*time.Second).Should(BeTrue(), "Curl debug pod did not reach Running state")
+
+	return nil
+}
+
+// Execute a curl command inside the debug pod
+func execCurlInPod(namespace string, cmd []string) (string, error) {
```

Comment:
can you reuse this method?
https://github.com/kaito-project/kaito/blob/main/test%2Fe2e%2Futils%2Futils.go#L164-L164

Reply:
done